### DISCUSSION TITLE
Code-first command?

### DISCUSSION THREAD

**Discussion by lorentmustafa** [bilal-fazlani/commanddotnet#484] [General]
Spending some time reading the documentation and experimenting with the tool.

Wonder if there's a way that, creating an object from code would return it into a command line string before passing it to a process runner. Or am i missing the idea?

Lets say:

[Command(Name:"a")]
public class A {
   [Option("x")
    public int X {get; set;}
   [Option("y")
    public int Y {get; set;}
}

and then the creation of an object var A = new A { x = 1, y = 2 }, would pass the command to the runner as # a -x 1 -y 2.

**Comment by bilal-fazlani** [comment]
When inside dotnet runtime, wouldn't it be easier to just call the methods directly?

**Reply by lorentmustafa** [reply]
Yeah i get the idea, im just making a wrapper for another console application i have to use inside dotnet runtime, and was thinking if i could make use of the library to interpret the properties in the correct format, without writing custom formatters.

**Comment by drewburlingame** [comment]
Are you looking to serialize the object to `a -x 1 -y 2` as output from the source application?  

With CommandDotNet, arguments are not defined directly on a command class, but rather are always pinned to a method, so it wouldn't know how to interpret what you've defined above.

If you're looking to pass the serialized object json to the target application, the target application would need a separate command to parse the json since it's a different format than the command being passed in.  Or you'd create a directive and token transformer that could convert the json into arguments before the command processor parses them.

Either way, there's nothing built in for this.
